### PR TITLE
GEODE-6043: Improve Auth example

### PR DESCRIPTION
**Pull request by mmartell**
- Implement end to end authentication

Co-authored-by: Ivan Godwin <igodwin@pivotal.io>

### CODE REVIEW

File: examples/utilities/CMakeLists.txt [hunk 1]
```diff
@@ -22,5 +22,8 @@ configure_file(${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt.in ${CMAKE_CURRENT_BIN
 install(FILES
   ${CMAKE_CURRENT_BINARY_DIR}/CMakeLists.txt
   ${CMAKE_CURRENT_SOURCE_DIR}/ExampleMultiGetFunction.java
+  ${CMAKE_CURRENT_SOURCE_DIR}/DummyAuthenticator.java
```
Comment (pivotal-jbarrett) [inline examples/utilities/CMakeLists.txt]: Dummy is a bit negative. Simple maybe?

File: examples/utilities/UsernamePrincipal.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+package javaobject;
+
+import java.io.Serializable;
+import java.security.Principal;
+
+/**
+ * An implementation of {@link Principal} class for a simple user name.
+ * 
+ * 
+ * @since 5.5
```
Comment (pivotal-jbarrett) [inline examples/utilities/UsernamePrincipal.java]: Since 5.5 what?

File: examples/utilities/DummyAuthenticator.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+import javaobject.UserPasswordAuthInit;
+import javaobject.UsernamePrincipal;
+
+/**
+ * A dummy implementation of the {@link Authenticator} interface that expects a
+ * user name and password allowing authentication depending on the format of the
+ * user name.
+ * 
+ * 
+ * @since 5.5
```
Comment (pivotal-jbarrett) [inline examples/utilities/DummyAuthenticator.java]: Since 5.5 what?

File: examples/utilities/UserPasswordAuthInit.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+/**
+ * An {@link AuthInitialize} implementation that obtains the user name and
+ * password as the credentials from the given set of properties.
+ * 
+ * To use this class the <c>security-client-auth-init</c> property should be
+ * set to the fully qualified name the static <code>create</code> function
+ * viz. <code>templates.security.UserPasswordAuthInit.create</code>
+ * 
+ * 
+ * @since 5.5
```
Comment (pivotal-jbarrett) [inline examples/utilities/UserPasswordAuthInit.java]: Since 5.5 what?

File: examples/utilities/UsernamePrincipal.java (hunk 2 above)
Comment (mmartell) [reply examples/utilities/UsernamePrincipal.java]: Removed as not necessary.

File: examples/utilities/UserPasswordAuthInit.java (hunk 4 above)
Comment (mmartell) [reply examples/utilities/UserPasswordAuthInit.java]: Removed as unneccessary.

File: examples/utilities/DummyAuthenticator.java (hunk 3 above)
Comment (mmartell) [reply examples/utilities/DummyAuthenticator.java]: Removed as unneccessary.

File: cppcache/src/ExecutionImpl.cpp [hunk 5]
```diff
@@ -131,7 +131,8 @@ std::shared_ptr<ResultCollector> ExecutionImpl::execute(
   serverOptimizeForWrite = ((attr->at(2) == 1) ? true : false);
 
   LOGDEBUG(
-      "ExecutionImpl::execute got functionAttributes from server for function = "
+      "ExecutionImpl::execute got functionAttributes from server for function "
+      "= "
```
Comment (pivotal-jbarrett) [inline cppcache/src/ExecutionImpl.cpp]: Dude, bad format here. The lines below can fit on this line. Please cleanup.

Comment (mmartell) [reply cppcache/src/ExecutionImpl.cpp]: Fixed. Good catch.

**Pull request merged**